### ISSUE TITLE
margin-horizontal property-no-unknown

### ISSUE THREAD

**Issue by sibelius** [styled-components/stylelint-processor-styled-components#134]
![image](https://user-images.githubusercontent.com/2005841/31504640-48fa986a-af49-11e7-9790-eeb3a731610e.png)

I'm getting this warning on react-native and styled-components

does it make sense?

same with padding-vertical

**Comment by emilgoldsmith** [comment]
Well.... Isn't it simply invalid CSS???

Do you want to write
```css
margin: 0 SOME_HORIZONTAL_VALUE
```

??

**Comment by emilgoldsmith** [comment]
That's what the linter is for :). Catching your CSS errors.

Here is the syntax for margin (and the one for padding is basically the same):

https://www.w3schools.com/Css/css_margin.asp

I'm gonna close this as this doesn't have anything to do with us.

**Comment by sibelius** [comment]
I guess this is valid on React Native

check this out: https://facebook.github.io/react-native/docs/layout-props.html#marginhorizontal

**Comment by emilgoldsmith** [comment]
Ahh okay, fair enough, I'm not very familiar with React Native. But does Stylelint support React Native?

Again, it's still the correct thing for me to close this because it doesn't have anything to do with us. This project simply parses Styled Components CSS and passes it to Stylelint.

I've never myself used Styled Components with React Native so if @mxstbr has some input that shows there maybe is something we should be supporting here feel free to give input. I doubt it though as in the end we can only support what Stylelint supports, maybe you need specific Stylelint rules, or use a different library than Stylelint. Again, I don't know much about React Native so what I'm saying may be garbage.

**Comment by mxstbr** [comment]
Yeah, ReactNative has a slightly different system of styling. I don't know if stylelint has support for that, if so you might have to use some sort of ReactNative plugin or something?

**Comment by sibelius** [comment]
I think we should reopen this until we figure it out how to properly handle react-native case

**Comment by emilgoldsmith** [comment]
I'm just not sure if this is in the realm of this repo? React Native isn't really CSS right? It's pseudo CSS as far as I know.

**Comment by mxstbr** [comment]
This unfortunately has nothing to do with this processor at all. All this processor does is turn this:

```JS
const View = styled.View`
  color: blue;
`
```

into this for stylelint:

```CSS
.selector {
  color: blue;
}
```

It doesn't do anything else. To specifically lint ReactNative styling you'd likely have to write a plugin or preset for stylelint that encompasses all of the valid rules etc.

I'm happy for you to tackle that! :blush:

**Comment by sibelius** [comment]
fixed with this .stylelintrc

```
{
  "processors": ["stylelint-processor-styled-components"],
  "extends": [
    "stylelint-config-standard",
    "stylelint-config-styled-components"
  ],
  "syntax": "scss",
  "rules": {
    "block-no-empty": null,
    "declaration-block-no-redundant-longhand-properties": null,
    "block-closing-brace-newline-before": null,
    "indentation": null,
    "property-no-unknown": [ true, {
      "ignoreProperties": [
        "padding-vertical",
        "padding-horizontal",
        "shadow-color",
        "shadow-offset",
        "shadow-radius",
        "shadow-opacity"
      ]
    }
    ]
  }
}
```

**Comment by mxstbr** [comment]
Ohh, that's a nice workaround! Is there a comprehensive list of ReactNative-specific properties? /cc @vjeux

**Comment by vjeux** [comment]
https://github.com/facebook/react-native/blob/master/Libraries/Text/TextStylePropTypes.js
https://github.com/facebook/react-native/blob/master/Libraries/Image/ImageStylePropTypes.js
https://github.com/facebook/react-native/blob/master/Libraries/Components/View/ViewStylePropTypes.js

**Comment by emilgoldsmith** [comment]
I guess this could lead to a us creating an extendable react native config for the processor? Like the main one @ismay created? That could be a pretty elegant easy thing to add anyway.

**Comment by mxstbr** [comment]
Yeah I like that, `stylelint-config-react-native`?

**Comment by emilgoldsmith** [comment]
Yeah that's a pretty clean name, I guess we don't even need to include Styled Components in the name as it's not directly related to it, which is much cleaner (`stylelint-config-styled-components-react-native` is pretty damn long :P).

Let's do it!
